File: CDK/Sources/Platform/android/java/src/kr/co/brgames/cdk/CSLocalNotification.java
package kr.co.brgames.cdk;

public class CSLocalNotification {
    public static void addNotification(final int id, final String title, final String message, final int delay) {
        CSActivity.sharedActivity().queueEvent(() -> {
            String titleOrName = title != null ? title : CSActivity.sharedActivity().resourceString("app_name");
            int smallIconId = CSActivity.sharedActivity().resourceAsId("@drawable/ic_c2s_notification_small_icon");

            CSNotification.addNotification(id, titleOrName, message, smallIconId, 0, delay);
        });
    }

    public static void removeNotification(final int id) {
        CSActivity.sharedActivity().queueEvent(() -> CSNotification.removeNotification(id));
    }

    public static void clearNotifications() {
        CSActivity.sharedActivity().queueEvent(CSNotification::clearNotifications);
    }
}

File: CDK/Sources/Platform/android/CSLocalNotification.cpp
//
//  CSNaverAuthLogin.cpp
//  CDK
//
//  Created by WooyolJung on 2017. 12. 13.
//  Copyright © 2017년 brgames. All rights reserved.
//

#ifdef CDK_ANDROID

#define CDK_IMPL

#include "CSLocalNotification.h"

#include "CSJNI.h"

void CSLocalNotification::addNotification(int notificationId, const char* title, const char* message, int delay) {
    CSJNIMethod mi;
    CSJNI::findMethod(mi, CSJNIClassLocalNotification, "addNotification", "(ILjava/lang/String;Ljava/lang/String;I)V", false);
    jstring jtitle = title ? mi.env->NewStringUTF(title) : NULL;
    jstring jmessage = mi.env->NewStringUTF(message);
    mi.env->CallStaticVoidMethod(mi.classId, mi.methodId, notificationId, jtitle, jmessage, delay);
    if (jtitle) mi.env->DeleteLocalRef(jtitle);
    mi.env->DeleteLocalRef(jmessage);
}

void CSLocalNotification::removeNotification(int notificationId) {
    CSJNIMethod mi;
    CSJNI::findMethod(mi, CSJNIClassLocalNotification, "removeNotification", "(I)V", false);
    mi.env->CallStaticVoidMethod(mi.classId, mi.methodId, notificationId);
}

void CSLocalNotification::clearNotifications() {
    CSJNIMethod mi;
    CSJNI::findMethod(mi, CSJNIClassLocalNotification, "clearNotifications", "()V", false);
    mi.env->CallStaticVoidMethod(mi.classId, mi.methodId);
}

#endif
